action.yml

```
@@ -72,45 +72,65 @@ runs:
       shell: bash
       run: |
         aws ecs update-service \
-          --health-check-grace-period-seconds 900 \
           --task-definition "${{ steps.deploy-task-definition.outputs.latest-task-definition-arn }}" \
+          --desired-count 1 \
           --cluster ${{ inputs.environment }} \
           --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
-          --desired-count 1 \
           --region ${{ inputs.aws_region }}
 
     - name: Await service stability
       shell: bash
-      # ToDo: MD-7199 Re-evaluate logic after deployment failures are correctly handled by ECS
       id: check-service-stability
       run: |
         aws ecs wait services-stable \
           --cluster ${{ inputs.environment }} \
           --service ${{ inputs.ecr_repository }}-${{ inputs.environment }} \
           --region ${{ inputs.aws_region }}
 
-    - name: Deregister previous local-exec task definition
-      if: ${{ always() && steps.deploy-local-task-definition.outputs.previous-task-definition-arn != '' }}
+    - name: Determine which task definition arn should be deregistered
+      if: always()
       shell: bash
+      id: determine-arn-to-deregister
```

```suggestion
      id: determine-task-definitions-to-deregister
```

I understand what you mean, but I think writing ARN ("Amazon Resource Name") here might be confusing to others. I think it should be "definitions" (plural), because you later want to deregister both the production and local-exec task definitions.